### PR TITLE
Add a target to build but not run tests

### PR DESCRIPTION
**Pull request by mulkieran**
Related: https://github.com/stratis-storage/project/issues/43.

### CODE REVIEW

**Comment by mulkieran** [comment]
Supercedes #1596 .

File: README.md [hunk 1]
```diff
@@ -62,6 +62,10 @@ Once toolchain and other dependencies are in place, run `make build` to build, a
 `stratisd` executable in `./target/x86_64-unknown-linux-gnu/debug` as root.
 Pass the `--help` option for more information on additional developer options.
 
+##### Building tests
+The Makefile provides a target, `build-tests` which allows compiling the
+tests without running any of them, as a convenience to developers.
```
Comment (GuillaumeGomez) [inline README.md]: Just wondering: shouldn't it be "as a convenience **for** developers"? If there is a rule/exception about this in english, I'd really like to learn it.

Comment (mulkieran) [reply README.md]: Either is correct. There's probably a very subtle difference in meaning. Something like "convenient to" would make more sense when thinking of a tool that could be grabbed quickly (probably my thought at the time). "convenient for" would emphasize the person and their experience more.

**Comment by mulkieran** [comment]
Whoops. Same error of pulling from wrong PR recurring. This definitely requires investigation now.

File: README.md (hunk 1 above)
Comment (GuillaumeGomez) [reply README.md]: I see, thanks for the explanation!

**Comment by mulkieran** [comment]
rebased.

**Pull request merged**